app/src/main/java/com/nervousfish/nervousfish/activities/RhythmCreateActivity.java

```diff
@@ -191,20 +193,14 @@ public void onStopRecordingClick(final View v) {
     }
 
     /**
-     * Called when a new data is received.
+     * Called when new bytes are received.
      *
-     * @param event Contains additional data about the event
+     * @param event Contains the bytes that are received
      */
     @Subscribe(threadMode = ThreadMode.MAIN)
-    public void onNewDataReceivedEvent(final NewDataReceivedEvent event) {
-        LOGGER.info("onNewDataReceivedEvent called");
-        if (event.getClazz().equals(Contact.class)) {
-            final Contact contact = (Contact) event.getData();
-            ContactReceivedHelper.newContactReceived(this.database, this, contact);
-
-            //This needs to be outside of the try catch block
-            this.dataReceived = contact;
-        }
+    public void onNewBytesReceivedEvent(final NewEncryptedBytesReceivedEvent event) {
```

Comment:
Can you rename this to `onEncryptedBytesReceivedEvent` to make it clear that this is about encrypted data?